apps/prairielearn/src/sprocs/instance_questions_points_homework.sql

```diff
@@ -69,7 +69,9 @@ BEGIN
     length := cardinality(variants_points_list);
     auto_points := 0;
     FOR i in 1..length LOOP
-        auto_points := auto_points + variants_points_list[i];
+        -- We convert values to `numeric` to avoid accumulating floating-point
+        -- errors when summing non-integer points.
+        auto_points := auto_points::numeric + variants_points_list[i]::numeric;
```

Comment:
This is fine but I think it would be even safer to declare a new `auto_points_accumulator` variable that has type `numeric` for use in the loop, then convert back to the double-precision `auto_points` only once at the end. At the moment we are forcing the accumulator back to double-precision at each loop iteration.

Reply:
I made this change in 9f3a619, how's that look?